source/common/router/scoped_rds.cc

```diff
@@ -164,10 +169,8 @@ ScopedRdsConfigSubscription::ScopedRdsConfigSubscription(
             *this, resource_decoder_);
   }
 
-  initialize([scope_key_builder]() -> Envoy::Config::ConfigProvider::ConfigConstSharedPtr {
-    return std::make_shared<ScopedConfigImpl>(
-        envoy::extensions::filters::network::http_connection_manager::v3::ScopedRoutes::
-            ScopeKeyBuilder(scope_key_builder));
+  initialize([]() -> Envoy::Config::ConfigProvider::ConfigConstSharedPtr {
```

Comment:
I'm not sure what's the benefit of having this callback now (no need to access the `scope_key_builder` on initialization of the object).
I don't think that this should be taken care of as part of this PR, but a TODO for investigation can be added here.